zkEVM: skeleton menu plus Overview section (#459)
0xPolygon/polygon-docs: docs/zkEVM/architecture/high-level/overview.md

<!-- https://excalidraw.com/#json=JKZp9QEihifF_B7Z41Dfv,FVNhqQKi9PA1jM0kzUoCsQ" -->

## Intro

> This section of the docs looks more closely at the full topology of Polygon's zero-knowledge system architecture. 

Reviewer: ```suggestion
This section provides a detailed analysis of Polygon's zero-knowledge system architecture, exploring its full topology.
```

Author: this was outdated, i included some of your suggestions

> Built for CDK stacks and using zkEVM technology, these documents explain the contents, layout, and interactions between component systems and functions, drilling down into detailed descriptions of the components that make up the wider system with reference to the relevant code bases.

Reviewer: ```suggestion
Tailored for CDK stacks and leveraging zkEVM technology, these documents explore the contents, layout, and interactions among component systems and functions. They offer in-depth descriptions of the components that constitute the larger system, with references to relevant code bases.
```

The diagram below is a full and detailed topological overview of the entire Polygon zero-knowledge system architecture.

## Architectural topology

![Polygon systems topology](../../../img/cdk/high-level-architecture/full-topology.png)

## Components 

> - Smart contracts: L1 and L2 Solidity smart contracts used for interacting with the whole stack. See the discussion on [zkEVM smart contracts](../../architecture/protocol/zkevm-bridge/smart-contracts.md) for more information.

Reviewer: ```suggestion
- Smart contracts: L1 and L2 Solidity smart contracts used for interacting with the network. See the discussion on [zkEVM smart contracts](../../architecture/protocol/zkevm-bridge/smart-contracts.md) for more information.
```

- Exit root trees: Append-only sparse Merkle trees which record the current and historical state of the system. See the discussion on [zkEVM exit trees](../../architecture/protocol/zkevm-bridge/exit-tree.md) for more information.

> - Aggregator: Used for aggregating transactions into batches for proving. See the discussion on [aggregators](../../architecture/index.md#aggregator) for more information.

Reviewer: ```suggestion
- Aggregator: Used for aggregating transaction batch sequences for proving. See the discussion on [aggregators](../../architecture/index.md#aggregator) for more information.
```

> - Sequencer: Does the complex job of carefully sequencing transactions as they come in before sending them to the aggregator for batching. See the discussion on [sequencers](../../architecture/index.md#sequencer) for more information.

Reviewer: ```suggestion
- Sequencer: Takes transactions from the transaction mempool and generates batches before sending them to the aggregator. See the discussion on [sequencers](../../architecture/index.md#sequencer) for more information.
```

> - Synchronizer: This component ensures a synchronized state between the node's systems and the L1 outside world via the Etherman component and the state database. 

Reviewer: This needs a more detailed description with specifics. Can still be described in one line with the necessary information.

Author: we are going into detail later, please make a suggestion ...

<!-- https://excalidraw.com/#json=JKZp9QEihifF_B7Z41Dfv,FVNhqQKi9PA1jM0kzUoCsQ" -->

## Intro

This section of the docs provides a detailed analysis of the full topology of the Polygon zero-knowledge system architecture. 

Tailored for CDK stacks and using zkEVM technology, these documents explore the contents, layout, and interactions among component systems and functions. They offer in-depth descriptions of the components that constitute the larger system, with references to relevant code bases.

The diagram below is a full and detailed topological overview of the entire Polygon zero-knowledge system architecture.

## Architectural topology

![Polygon systems topology](../../../img/cdk/high-level-architecture/full-topology.png)

## Components 

> - Smart contracts: L1 and L2 Solidity smart contracts used for interacting with the whole stack. See the discussion on [zkEVM smart contracts](../../architecture/protocol/zkevm-bridge/smart-contracts.md) for more information.

Author: I don't think the Solidity facilitates faster transaction and scalability, that would be the core processing.

- Exit root trees: Append-only sparse Merkle trees which record the current and historical state of the system. See the discussion on [zkEVM exit trees](../../architecture/protocol/zkevm-bridge/exit-tree.md) for more information.
- CDK and zkEVM nodes containing:

>     - Aggregator: Used for aggregating transactions into batches for proving. See the discussion on [aggregators](../../architecture/index.md#aggregator) for more information.

Author: ```suggestion
    - Aggregator: Used for aggregating transaction batches to send to the prover. See the discussion on [aggregators](../../architecture/index.md#aggregator) for more information.
```

    - Pool database: The pool database records transaction requests coming in from the JSON RPC client and sends them to the sequencer.
    - State database: The state database responds to read requests from the JSON RPC client.
    - Sequencer: Does the complex job of carefully sequencing transactions as they come in before sending them to the aggregator for batching. See the discussion on [sequencers](../../architecture/index.md#sequencer) for more information.
    - Aggregator: Used for aggregating transaction batches to send to the prover. See the discussion on [aggregators](../../architecture/index.md#aggregator) for more information.

>     - Synchronizer: This component ensures a synchronized state between the node's systems and the L1 outside world via the Etherman component and the state database. 

Reviewer: ```suggestion
    - Synchronizer: Supports co-ordination between contracts and off-chain zkEVM execution environment ensuring integrity of state transitions via the Etherman component and the state database.
```

>     - State database: The state database responds to read requests from the JSON RPC client.

Reviewer: This definition needs to briefly cover what state information is stored in the DB and how it is used by the network.

Author: we are going into detail later

> - Bridge service component: Main facility for transferring tokens from L1 to L2 and back again.

Reviewer: ```suggestion
- Bridge service component: Main facility for transferring tokens between L1 and L2.
```

- Prover component: System for calculating zero-knowledge proofs on transaction batches.

## What to expect

When complete, this section will include information on: 

- The structure of a CDK node and how it interacts with L2 and L1 smart contracts.
- The structure of a zkEVM node and how it interacts with L2 and L1 smart contracts.

> - The key components included in the nodes, how they function, and their interactions with other components, external developers, and the L1/L2 environment.

Reviewer: ```suggestion
- The key components included in the nodes, how they function, and their interactions with other components, external dApps, and the L1/L2 environment.
```

> - Key similarities and differences between a CDK and zkEVM node set up.

Reviewer: ```suggestion
- Key similarities and differences between a CDK and zkEVM node set-up.
```
